### PR TITLE
New website with minimal changes required

### PR DESCRIPTION
**Pull request by wtrocki**
1. I have put docs as front page for now (better design will follow in separate PR)
2. Minimized changes in the files (although some structure changes were needed)
3. Left Markdown readme functional (for the moment) so github can also be used to view all content

Published to: https://nodeshift.dev/nodejs-reference-architecture/

### CODE REVIEW

**Comment by wtrocki** [comment]
@mhdawson reason why website took so long is that introducing it caused multiple conflicts. 
Would it be possible to fastract review on this so it will not happen (This is thrid attempt on website :))

File: website/docs/development/npm-publishing.md [hunk 1]
```diff
@@ -1,17 +1,20 @@
-# Npm Publishing Guidelines
+---
+sidebar_position: 4
+---
 
+# Npm Publishing Guidelines
 
 ## Guidance
 
-When publishing modules to npm, it is recommened to enable 2 factor authenticaton.  How to set that up can be found here https://docs.npmjs.com/configuring-two-factor-authentication.  It is also a good practice to require the use of two-factor authentication or automation tokens for each module that is published.
```
Comment (helio-frota) [inline website/docs/development/npm-publishing.md:6]: nit recommended
nit authentication

File: website/docs/development/npm-publishing.md [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+When publishing modules to npm, it is recommened to enable 2 factor authenticaton. How to set that up can be found here https://docs.npmjs.com/configuring-two-factor-authentication. It is also a good practice to require the use of two-factor authentication or automation tokens for each module that is published.
 
 If the module will be maintained by a group of people, it is better to create an organization for your team/project on npm to make it easier for multiple maintainers to manage and publish the package.
 
-It is also a good practice to have a "prepublish" npm script in the package.json that runs linting and tests before publishing happens.  If your code needs to be transpiled first, having a "prepublish" script will make sure this is also done before publishing.
+It is also a good practice to have a "prepublish" npm script in the package.json that runs linting and tests before publishing happens. If your code needs to be transpiled first, having a "prepublish" script will make sure this is also done before publishing.
 
 The use of scoped pacakges is a good practice if you are creating many related packages.
```
Comment (helio-frota) [inline website/docs/development/npm-publishing.md:15]: nit packages

File: website/docs/development/npm-publishing.md [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
 
-When creating modules that will be published to an internal registry that is not upstream npm, it is a good practice to scope those modules to the name of your organization.  It is also important that you or your company are the owners of the scoped orginization on upstream npm.  This is important if a public module which depends on a private module is accidently published to upstream npm, which can lead to the "Dependecy Confusion" hack that is outlined [in this post](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610)
+When creating modules that will be published to an internal registry that is not upstream npm, it is a good practice to scope those modules to the name of your organization. It is also important that you or your company are the owners of the scoped orginization on upstream npm. This is important if a public module which depends on a private module is accidently published to upstream npm, which can lead to the "Dependecy Confusion" hack that is outlined [in this post](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610)
```
Comment (helio-frota) [inline website/docs/development/npm-publishing.md:17]: nit scoped organization

Comment (helio-frota) [reply website/docs/development/npm-publishing.md:17]: nit Dependency Confusion

Comment (helio-frota) [reply website/docs/development/npm-publishing.md:17]: nit accidentally

File: website/docs/development/npm-publishing.md [hunk 4]
```diff
@@ -29,21 +32,21 @@ Which one to go with is solely up to you and your preference.
...
+- exclude tests
 
+Not including these files can decrease the size of the package that is being installed. If the source code for the package is hosted on github, it is a common practice to use github pages to host any API reference docs and provide a link to those in the README and the package.json.
 
 #### Transpiling Sources
 
 A package written in another language like TypeScript should be published with transpiled JavaScript as the primary executable.
 
-Generally the original sources should not be required to use your published package, but it's your preference whether to publish them or not.  Similar to the guidance on tests, the modules that the team publishs that are transpiled only contain the generated code.
+Generally the original sources should not be required to use your published package, but it's your preference whether to publish them or not. Similar to the guidance on tests, the modules that the team publishs that are transpiled only contain the generated code.
```
Comment (helio-frota) [inline website/docs/development/npm-publishing.md:45]: nit publishes

File: website/docs/development/npm-publishing.md [hunk 5]
```diff
@@ -55,25 +58,24 @@ It is important if you are using Typescript or are generating type definitions f
 
 #### Module Versions
 
-It is recommened to use [Semantic Versioning](https://semver.org/) when possible.  This will make it easier for a user to determine if a new version of a module will potentianlly break their code.
+It is recommened to use [Semantic Versioning](https://semver.org/) when possible. This will make it easier for a user to determine if a new version of a module will potentianlly break their code.
```
Comment (helio-frota) [inline website/docs/development/npm-publishing.md:61]: nit recommended

Comment (helio-frota) [reply website/docs/development/npm-publishing.md:61]: nit potentially

File: website/docs/development/npm-publishing.md [hunk 6]
```diff
@@ -55,25 +58,24 @@ It is important if you are using Typescript or are generating type definitions f
...
 
 There are some automation tools that can help with bumping your module to the correct version.
 
-* [release-please](https://github.com/googleapis/release-please)
-* [standard-version](https://github.com/conventional-changelog/standard-version)
+- [release-please](https://github.com/googleapis/release-please)
+- [standard-version](https://github.com/conventional-changelog/standard-version)
 
 Both of those tools will base the version bump on commit messages that follow the [coventional commit standard](https://www.conventionalcommits.org/en/v1.0.0/)
```
Comment (helio-frota) [inline website/docs/development/npm-publishing.md:68]: nit conventional

File: website/docs/development/npm-publishing.md [hunk 7]
```diff
@@ -55,25 +58,24 @@ It is important if you are using Typescript or are generating type definitions f
...
 
 The npm cli also provides a [`version` command](https://docs.npmjs.com/cli/v7/commands/npm-version) that can be used to increase the version of package.
 
-
 #### Publish Internal Modules
 
-When publishing to an internal registry, like Artifactory or Nexus, it is recommended to scope your package.  This adds extra protection if this module is accidently published to a public registry and avoids the dependecy confusion hack mentioned above.
+When publishing to an internal registry, like Artifactory or Nexus, it is recommended to scope your package. This adds extra protection if this module is accidently published to a public registry and avoids the dependecy confusion hack mentioned above.
```
Comment (helio-frota) [inline website/docs/development/npm-publishing.md:74]: nit accidentally

Comment (helio-frota) [reply website/docs/development/npm-publishing.md:74]: nit dependency

Comment (helio-frota) [reply website/docs/development/npm-publishing.md:74]: nit recommends

**Review by helio-frota — CHANGES_REQUESTED** [review]

**Comment by wtrocki** [comment]
@helio-frota thank you for review. I will  apply those suggestions in separate PR. Trying to not make any content changes in this PR. it is only for website publishing

**Comment by helio-frota** [comment]
@wtrocki  :+1:

**Review by helio-frota — APPROVED** [review]

**Comment by wtrocki** [comment]
Thank you so much Helio!

**Review by BethGriggs — APPROVED** [review]

**Review by lholmquist — APPROVED** [review]

**Comment by mhdawson** [comment]
Had short discussion with @wtrocki today. He is going to refactor so that we don't have to move the content to website/docs in the repo. This will avoid breaking existing links and I think will be handled as a copy during the build.

Other than that the changes to the existing docs look like they are only formatting/fixup so look good to me.

File: docs/development/testing.md [hunk 8]
```diff
@@ -135,18 +139,18 @@ Mocha does not automatically compute code coverage. If you need it, use:
 - [nyc][]: the most popular code-coverage tool; the successor CLI for Istanbul
 
 [`assert` module]: https://nodejs.org/api/assert.html#assert_assert
-[Babel]: https://babeljs.io
-[Chai]: https://www.npmjs.com/package/chai
+[babel]: https://babeljs.io
```
Comment (wtrocki) [inline docs/development/testing.md:142]: Ok.. So I have run lint but apparently, that doesn't seem to work well with this style of links
Fixing it manually - there were multiple lint issues checked on CI/CD

File: docs/development/apifirst.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (wtrocki) [inline docs/development/apifirst.md:1]: Removing it.

**Comment by wtrocki** [comment]
Good review

**Review by mhdawson — APPROVED** [review]
LGTM

**Pull request merged**